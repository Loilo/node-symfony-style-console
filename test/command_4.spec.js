

const wrapper = require('./wrapper')

test('04 Ensure has single blank line after any text and a title', wrapper(io => {
    io.write('Lorem ipsum dolor sit amet')
    io.title('First title')

    io.writeln('Lorem ipsum dolor sit amet')
    io.title('Second title')

    io.write('Lorem ipsum dolor sit amet')
    io.write('')
    io.title('Third title')

    // Ensure edge case by appending empty strings to history:
    io.write('Lorem ipsum dolor sit amet')
    io.write(['', '', ''])
    io.title('Fourth title')

    // Ensure have manual control over number of blank lines:
    io.writeln('Lorem ipsum dolor sit amet')
    io.writeln(['', '']) //Should append an extra blank line
    io.title('Fifth title')

    io.writeln('Lorem ipsum dolor sit amet')
    io.newLine(2) //Should append an extra blank line
    io.title('Fifth title')
}))
