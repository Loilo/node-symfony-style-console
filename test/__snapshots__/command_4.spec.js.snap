// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`04 Ensure has single blank line after any text and a title 1`] = `
"Lorem ipsum dolor sit amet

First title
===========

Lorem ipsum dolor sit amet

Second title
============

Lorem ipsum dolor sit amet

Third title
===========

Lorem ipsum dolor sit amet

Fourth title
============

Lorem ipsum dolor sit amet


Fifth title
===========

Lorem ipsum dolor sit amet


Fifth title
===========

"
`;
